### ISSUE TITLE
Using local model with torchub on GPU

### ISSUE THREAD

**Issue by Squeemos** [ultralytics/yolov5#5005]
## ❔Question
So I finished training my model so my plan was to start using it in my application. However, loading it the way I'm currently doing means that the inference time is very slow and nowhere near as fast as it should be.

The basics of my script are this:
```
model = torch.hub.load('./yolov5','custom',path = './yolov5/runs/train/exp16/weights/best.pt',source='local')
model.cuda()

while True:
    # Take a screenshot with mss and feed it into the model

    start_time = time.time()
    result = model(new_img,size=IMG_SHAPE[0])

    result.print()
    print(result.pandas().xyxy[0])

    print(f"Time taken: {(time.time() - start_time)}",end="\r")
```

I tried moving the taking a screenshot outside of the loop to see if that would speed things up but it didn't (makes sense since the image processing aspect of that is really fast).
Is there something I can do to speed up the inference time? Apart from using the detect.py since I need to be able to read out the labels in my own script.

## Additional context
With taking a screenshot every iteration
![With taking a screenshot every iteration](https://user-images.githubusercontent.com/56849503/135391346-0ff94aa9-d322-4646-abb7-5354cd703248.png)

Without taking a screenshot every iteration
![Without taking a screenshot every iteration](https://user-images.githubusercontent.com/56849503/135391562-e4f2a2df-f88f-4e7e-9488-5044becbebf1.png)

I used the YOLOv5x pretrained weights to train my data from.

**Comment by glenn-jocher** [comment]
@Squeemos 👋 Hello! Thanks for asking about **inference speed issues**. YOLOv5 🚀 can be run on CPU (i.e. `--device cpu`, slow) or GPU if available (i.e. `--device 0`, faster). You can determine your inference device by viewing the YOLOv5 line printed to screen:

<img width="985" alt="Screen Shot 2021-10-05 at 2 03 41 PM" src="https://user-images.githubusercontent.com/26833433/136102519-400d1468-590a-44af-b587-929e48b13e93.png">

If you would like to increase your inference speed some options are:

- Increase batch size ([YOLOv5 PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models only)
- Reduce `--img-size`, i.e. 1280 -> 640 -> 320
- Reduce model size, i.e. YOLOv5x -> YOLOv5l -> YOLOv5m -> YOLOv5s -> YOLOv5n
- Use a faster GPUs, i.e.: P100 -> V100 -> A100
- Use a free GPU backends with up to 16GB of CUDA memory: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
